lms/djangoapps/course_api/views.py

```diff
@@ -2,52 +2,167 @@
 Course API Views
 """
 
-from rest_framework.views import APIView
-from rest_framework.response import Response
-from rest_framework.reverse import reverse
+from rest_framework.exceptions import NotFound
+from rest_framework.views import APIView, Response
 
+from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import CourseKey
+
+
 from openedx.core.lib.api.view_utils import view_auth_classes
-from xmodule.modulestore.django import modulestore
+
+from .api import course_detail, list_courses
 
 
 @view_auth_classes()
-class CourseView(APIView):
+class CourseDetailView(APIView):
     """
-    Course API view
+    **Use Cases**
+
+        Request information on a course
+
+    **Example Requests**
+
+        GET /api/courses/v1/courses/{course_key_string}/
```

Comment:
Nit: "course_key_string" is an implementation detail parameter name.  Consider changing it to "course_id" or "course_key" (whatever other APIs use?) in this docstring.